packages/mendel-middleware/index.js

```diff
@@ -106,7 +106,17 @@ function MendelMiddleware(opts) {
         var pack = bpack({raw: true, hasExports: true});
         var decodedResults = trees.findTreeForHash(params.bundle, params.hash);
         if (!decodedResults || decodedResults.error) {
-            return notFound(res, decodedResults && decodedResults.error);
+            return notFound(res, decodedResults && decodedResults.error, params);
+        }
+
+        var modules = indexedDeps(decodedResults.deps.filter(Boolean));
+
+        if (!modules.length) {
+            // Something wrong, modules shouldn't be zero
+            return notFound(res, {
+                code: 'EMPTYBUNDLE',
+                message: 'Tree contents are empty'
+            }, params);
```

Comment:
Hey! I just realized something: You are applying `shasum` on the filename, right? You are not using `shasum` of the source code, correct? This might mean you have collisions even with `shasum`. When you have circular dependencies you can get zero entries added to the bundle because the infinite loop is prevented by the cache, but you never read the line that actually adds the bundle to the array.

Try removing shasum from `mendel-generator-prune` as explained in the other issue.